cmd/ocsp-updater/main.go

```diff
@@ -23,20 +28,74 @@ import (
 	"github.com/letsencrypt/boulder/sa"
 )
 
-// FatalError indicates the updater should stop execution
-type FatalError string
-
-func (e FatalError) Error() string { return string(e) }
-
 // OCSPUpdater contains the useful objects for the Updater
 type OCSPUpdater struct {
 	stats statsd.Statter
 	log   *blog.AuditLogger
-	cac   rpc.CertificateAuthorityClient
+	clk   clock.Clock
+
 	dbMap *gorp.DbMap
+
+	rpcMu *sync.RWMutex
```

Comment:
If it works, its fine. Without having looked at the code, I'm worried that there is race between the client get a failure notice and the server reconnecting after that occurs.

I assume the code is smarter than that, though.